### PR TITLE
Move CI / PR triggers to YAML files

### PR DESCRIPTION
**Pull request by jaredpar**
Azure Pipelines now supports us putting the branches which trigger PR
and CI validation into the YAML files. This change puts the current
contents of our build definition into the YAML files and updates our
creating feature branches document.

Note: i have not yet changed our build definition to read this
information. Want to commit this, give it a day to flow to the branches
then I will flip our definition over.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC  @dotnet/roslyn-infrastructure for review

**Comment by jasonmalinowski** [comment]
@jaredpar So how does this work? The branch configuration is read out of one branch (where is that configured) and the actual jobs themselves out of that branches file? It's bizarre then that it's the same file?

File: docs/infrastructure/feature branches azure.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+Note: these steps assume the remote `origin` points to the official [roslyn repository](https://github.com/dotnet/roslyn).
+
+``` cmd
+> git fetch origin
+> git checkout -B init origin/master
+> git push origin init:features/mono
+```
+
+## Adding branch to Azure Pipelines
+The following files need to be editted in order for GitHub to trigger Azure Pipelines Test runs on PRs:
```
Comment (jasonmalinowski) [inline docs/infrastructure/feature branches azure.md]:
```suggestion
The following files need to be edited in order for GitHub to trigger Azure Pipelines Test runs on PRs:
```

**Comment by jaredpar** [comment]
@jasonmalinowski 

> So how does this work? The branch configuration is read out of one branch (where is that configured) and the actual jobs themselves out of that branches file? It's bizarre then that it's the same file?

The way it works is this: The build definition in Azure is configured to look at a specific file. On every PR or commit to the repository the build definition will look at the file in it's current state. If the branch which was just changed is listed in the file then the build definition will trigger, else it will not.

**Comment by jasonmalinowski** [comment]
Alright, that makes more sense.

**Pull request merged**